cgi-bin/DW/Controller/Media.pm

```diff
@@ -63,14 +73,38 @@ sub media_bulkedit_handler {
 
         if ( $post_args->{"action:edit"} ) {
             my %post = %{$post_args->as_hashref||{}};
-            while ( my ($key, $secval) = each %post ) {
-                next unless $key =~ m/^security-(\d+)/;
-                my $mediaid = $1 >> 8;
-                my $media = DW::Media->new( user => $rv->{u}, mediaid => $mediaid );
+
+            # transform our HTML field names to property names
+            # and group by what media object they belong to
+            # we don't care if the id or prop might not exist
+            # right now, later steps will verify them
+
+            my %props;
+            while ( my ($key, $val) = each %post ) {
+                next unless $key =~ m/^(\w*)-(\d+)/;
```

Comment:
I think this should be \w+, not \w* - otherwise you risk $1 being undefined below.